[SPARK-45575][SQL] Support time travel options for df read API (#43403)
apache/spark: docs/sql-error-conditions-invalid-time-travel-timestamp-expr-error-class.md

Must be deterministic.

## OPTION

> Timestamp string in the options should be in the format of 'yyyy-MM-dd HH:mm:ss[.us][zone_id]'.

Reviewer: ```suggestion
Timestamp string in the options should use ISO standard format such as 'yyyy-MM-dd HH:mm:ss'.
```
